hack/test-cmd.sh

```diff
@@ -352,12 +345,19 @@ for test in "${tests[@]}"; do
   # switch back to a standard identity. This prevents individual tests from changing contexts and messing up other tests
   oc project ${CLUSTER_ADMIN_CONTEXT}
   oc new-project "cmd-${name}"
-  ${test}
+  if ! ${test}; then
+    failed="true"
+    tail -40 "${LOG_DIR}/openshift.log"
+  fi
   oc project ${CLUSTER_ADMIN_CONTEXT}
   oc delete project "cmd-${name}"
   cp ${KUBECONFIG}{.bak,}  # since nothing ever gets deleted from kubeconfig, reset it
 done
 
+if [[ -n "${failed:-}" ]]; then
```

Comment:
From the [manual](http://tldp.org/LDP/abs/html/comparison-ops.html):
```
7.3. Other Comparison Operators
string comparison
-n
    string is not null.
```

Quick example to show what that means for us:
```sh
#!/bin/bash

set -o errexit
set -o nounset
set -o pipefail

# run the test with no variable set
echo -n "\$value unset    : "
if [[ -n "${value:-}" ]]; then
	echo "true"
else
	echo "false"
fi

# run the test with an empty variable
echo -n "\$value=''       : "
value=''
if [[ -n "${value:-}" ]]; then
	echo "true"
else
	echo "false"
fi

# run the test with a non-empty variable
echo -n "\$value=\"test\"   : "
value="test"
if [[ -n "${value:-}" ]]; then
	echo "true"
else
	echo "false"
fi

# output
# $value unset    : false
# $value=''       : false
# $value="test"   : true
```

So this code will set `$failed` to  a value when a test fails, ergo the check here will fail.